metrics/metrics_cli.go

```diff
@@ -0,0 +1,191 @@
+package metrics
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gopkg.in/yaml.v3"
+)
+
+type Config struct {
+	Host string `yaml:"Host"`
+	Port string `yaml:"Port"`
+
+	Type      string `yaml:"Type"`
+	Count     int    `yaml:"Count"`
+	Frequency int    `yaml:"Frequency"`
+	Random    bool   `yaml:"Random"`
+}
+
+/*
+	defaultType valid values include - "all" "counter" "gauge" "histogram" "summary"
+	defaultCount valid values should be >= 0
+	defaultFreq valid values should be >= 0
+	defaultRand valid values should be boolean
+*/
+var defaultType = "all"
+var defaultCount = 1
+var defaultFreq = 15
+var defaultRand = false
+
+type CommandLine struct{}
+
+func (c *Config) Parse(data []byte) error {
+	return yaml.Unmarshal(data, c)
+}
+
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "healthy")
+}
+
+// initConnection handles the metric creation and also updates the metrics via go routines
+// The delegation logic is handled here
+func (cli *CommandLine) initConnection(metricType string, count int, freq int, isRandom bool, conf Config, address string) {
+
+	rand.Seed(time.Now().Unix())
+	mc := newMetricCollector()
+	mc.interval = time.Duration(freq)
+	switch metricType {
+	case "counter":
+		cli.createCounter(count, mc)
+	case "gauge":
+		cli.createGauge(count, mc)
+	case "histogram":
+		cli.createHistogram(count, mc)
+	case "summary":
+		cli.createSummary(count, mc)
+	case "all":
+		cli.createAll(count, mc, isRandom)
+	default:
+		log.Fatal("Invalid type")
+	}
+	log.Println("Serving on address: " + address)
+	if isRandom {
+		log.Println("Producing randomized metrics per type")
+
+	} else {
+		log.Println("Producing " + fmt.Sprintf("%d", count) + " metric(s) per type")
+	}
+	log.Println("Updating at a frequency of " + fmt.Sprintf("%d", mc.interval) + " seconds")
+	http.HandleFunc("/", healthCheckHandler)
+	http.Handle("/metrics", promhttp.HandlerFor(promRegistry, promhttp.HandlerOpts{}))
+
+	log.Fatal(http.ListenAndServe(address, nil))
```

Comment:
I'd suggest looking at ways to gracefully shut down the HTTP server when commanded by an OS signal.  See https://medium.com/honestbee-tw-engineer/gracefully-shutdown-in-go-http-server-5f5e6b83da5a for an example.  A separate issue can be created to handle this.